zp-relayer/workers/directDepositWorker.ts

```diff
@@ -0,0 +1,103 @@
+import { Job, Worker } from 'bullmq'
+import { logger } from '@/services/appLogger'
+import { DirectDeposit, DirectDepositResult } from '@/queue/directDepositQueue'
+import { DIRECT_DEPOSIT_QUEUE_NAME } from '@/utils/constants'
+import { buildPrefixedMemo, withErrorLog, withMutex } from '@/utils/helpers'
+import { pool } from '@/pool'
+import { sentTxQueue } from '@/queue/sentTxQueue'
+import { processDirectDeposits } from '@/txProcessor'
+import config from '@/configs/relayerConfig'
+import { TxValidationError, validateDirectDeposit } from '@/validateTx'
+import { IDirectDepositWorkerConfig } from './workerTypes'
+
+export async function createDirectDepositWorker({ redis, mutex, txManager }: IDirectDepositWorkerConfig) {
+  const workerLogger = logger.child({ worker: 'pool' })
+  const WORKER_OPTIONS = {
+    autorun: false,
+    connection: redis,
+    concurrency: 1,
+  }
+
+  const directDepositProcessor = async (job: Job<DirectDeposit[], DirectDepositResult>) => {
+    const jobLogger = workerLogger.child({ jobId: job.id })
+
+    const directDeposits = job.data
+    jobLogger.info('Received direct deposit', { number: directDeposits.length })
+
+    const validatedDeposits: DirectDeposit[] = []
+    for (const dd of directDeposits) {
+      try {
+        await validateDirectDeposit(dd)
```

Comment:
Does it mean that if validation fails for some deposits, we end up sending only a half-filled batch, even if there are other pending deposits for filling it?

Reply:
Yes, it is true. I tried to make it more optimal, but there is a trade-off. We either validate DD right before sending it and sometimes see unfilled batch (as implemented now) or we validate DD before adding a batch to the queue in watcher instance. Second approach will allow us to produce more optimal batches, but there will be a potential risk of an account becoming invalid/"bad".

Reply:
Right now we first select deposits for batch from the queue, and only then validate then. If we flip the order, and will validate deposits before forming a batch, it would work better.

Reply:
But currently the work of processing batches is done by watcher. And there could be a delay up to several hours(?) between making a batch and processing it. I assumed that during this time account could become invalid.

Reply:
Yes, but this does not imply that worker should send partially-filled batches, if it can send a full one

Reply:
Agreed internally to keep it as-is for now, to simplify the overall architecture